extensions/wikia/Email/Controller/CategoryAddController.class.php

```diff
@@ -58,6 +65,41 @@ protected function assertValidPageAddedToCategory() {
 		}
 	}
 
+	public function assertCanEmail() {
+		parent::assertCanEmail();
+		$this->assertEmailsNotThrottled();
+	}
+
+	private function assertEmailsNotThrottled() {
+		global $wgMemc;
+		$emailsSent = $wgMemc->get( $this->getTargetUserCacheKey() );
+		if ( $emailsSent != false && $emailsSent['sent'] >=  self::EMAILS_PER_THROTTLE_PERIOD ) {
+			throw new Check( 'Attempt to send too many emails' );
+		}
+	}
+
+	protected function afterSuccess() {
+		$this->incrementEmailsSentCount();
+	}
+
+	private function incrementEmailsSentCount() {
+		global $wgMemc;
+		$emailsSent = $wgMemc->get( $this->getTargetUserCacheKey() );
+		if ( $emailsSent == false ) {
+			$emailsSent = [
+				'sent' => 0,
+				'ttl' => time() + self::THROTTLE_PERIOD
+			];
+		}
+		$emailsSent['sent']++;
+
+		$wgMemc->set( $this->getTargetUserCacheKey(), $emailsSent, $emailsSent['ttl'] );
```

Comment:
You're right, I did some testing and that doesn't affect ttl. That's a cleaner solution.